packages/0018-package-format.md

```diff
@@ -0,0 +1,163 @@
+
+If the package you install or update has one or more package migrations that haven't been run you will need to authorize them with an admin account. This accomplishes two things - first is that you would have more control over things being run on the database (file changes are a bit easier to manage using version/source control such as Git).
+
+We want to show some sort of summary of what the migrations will do and present them to the user before they sign off on it.
+
+Opting in for migrations will also allow the user to plan when these things run. Can help with slow startup times due to migrations automatically running on startup.
+
+#### Warn about what will be removed on uninstall
+
+One thing that we want to see improved is the process of uninstalling a package. Currently, it can mess up your site if you uninstall a package you have used for content. Right now when you install a package it saves a list of all files and schema and content that it installed initially, and when you then uninstall it removes it all.
+
+So if you start with a starter kit and then build upon document types to make a site, and then remove the package all content built on the starter kit document types would disappear.
+
+We want to make it safer and more transparent what happens when you uninstall a package, we have several ideas but it became a huge project on its own. So, for now, the main idea is to show the user a list of things that will be removed when they try to uninstall. See more ideas under the Out of Scope section below.
+
+### Other
+
+#### NuGet feed on Our
+
+We will create a NuGet feed on the Our website for packages uploaded to Our. This means any package in the new .nupkg package format that is uploaded to Our will be able to be consumed by a new custom NuGet feed. This custom NuGet feed can be consumed within Visual Studio or any development IDE by adding the Our NuGet feed URL to your NuGet.config file. The feed URL is yet to be determined.
+
+If developers wish to upload their packages to NuGet as well that is totally ok. Perhaps in the future there will be a way to automatically publish a package from Our to your NuGet account. There's no problem with having the same package/version in 2 different feeds.
+
+## Drawbacks
+
+Discuss any disadvantages or sideeffects of this RFC.
+
+## Alternatives
+
+- Use proprietary format for backoffice creation, installation and uninstallation, similar to what exists.
+
+## Out of Scope
+
+We have a few ideas that would be awesome to have but will be out of scope for this RFC:
+
+- Automatically check for upgrades and notify users with the right permissions
+- Check for references in other content / schema when trying to uninstall something and warn the user
```

Comment:
Or maybe even force the user to manually remove content / schema which is referencing that which is to be uninstalled, before uninstall is even possible.